src/aap_eda/api/views/activation.py

```diff
@@ -88,6 +95,20 @@ def create(self, request):
             status=status.HTTP_201_CREATED,
         )
 
+    def perform_create(self, serializer):
+        instance = serializer.save()
+
+        if instance.credentials:
```

Comment:
@hsong-rh Can this be made into a function since we would be calling it only if we have some internal encrypted fields to use. E.g in fanout we have an extra var called EDA_PG_NOTIFY_DSN which has a secret embedded in it in that scenario we will want to create a system credential and then encrypt the EDA_PG_NOTIFY_DSN

Reply:
@mkanoor Based on the discussion, this logic is removed from here and will be added to the fanout project.